pkg/kubectl/cmd/edit.go

```diff
@@ -0,0 +1,419 @@
+
+The files to edit will be output in the default API version, or a version specified
+by --output-version. The default format is YAML - if you would like to edit in JSON
+pass -o json.
+
+In the event an error occurs while updating, a temporary file will be created on disk
+that contains your unapplied changes. The most common error when updating a resource
+is another editor changing the resource on the server. When this occurs, you will have
+to apply your changes to the newer version of the resource, or update your temporary
+saved copy to include the latest resource version.`
+
+	editExample = `  // Edit the service named 'docker-registry':
+  $ kubectl edit svc/docker-registry
+
+  // Edit the DeploymentConfig named 'my-deployment':
+  $ kubectl edit dc/my-deployment
+
+  // Use an alternative editor
+  $ OSC_EDITOR="nano" kubectl edit dc/my-deployment
+
+  // Edit the service 'docker-registry' in JSON using the v1beta3 API format:
+  $ kubectl edit svc/docker-registry --output-version=v1beta3 -o json`
+)
+
+var errExit = fmt.Errorf("exit directly")
+
+func NewCmdEdit(f *cmdutil.Factory, out io.Writer) *cobra.Command {
+	var filenames util.StringList
+	cmd := &cobra.Command{
+		Use:     "edit (RESOURCE/NAME | -f FILENAME)",
+		Short:   "Edit a resource on the server",
+		Long:    editLong,
+		Example: fmt.Sprintf(editExample),
+		Run: func(cmd *cobra.Command, args []string) {
+			err := RunEdit(f, out, cmd, args, filenames)
+			if err == errExit {
+				os.Exit(1)
+			}
+			cmdutil.CheckErr(err)
+		},
+	}
+	usage := "Filename, directory, or URL to file to use to edit the resource"
+	kubectl.AddJsonFilenameFlag(cmd, &filenames, usage)
+	cmd.Flags().StringP("output", "o", "yaml", "Output format. One of: yaml|json.")
+	cmd.Flags().String("output-version", "", "Output the formatted object with the given version (default api-version).")
+	return cmd
+}
+
+func RunEdit(f *cmdutil.Factory, out io.Writer, cmd *cobra.Command, args []string, filenames util.StringList) error {
+	var printer kubectl.ResourcePrinter
+	var ext string
+	switch format := cmdutil.GetFlagString(cmd, "output"); format {
+	case "json":
+		printer = &kubectl.JSONPrinter{}
+		ext = ".json"
+	case "yaml":
+		printer = &kubectl.YAMLPrinter{}
+		ext = ".yaml"
+	default:
+		return cmdutil.UsageError(cmd, "The flag 'output' must be one of yaml|json")
+	}
+
+	cmdNamespace, err := f.DefaultNamespace()
+	if err != nil {
+		return err
+	}
+
+	mapper, typer := f.Object()
+	rmap := &resource.Mapper{
+		ObjectTyper:  typer,
+		RESTMapper:   mapper,
+		ClientMapper: f.ClientMapperForCommand(),
+	}
+
+	b := resource.NewBuilder(mapper, typer, f.ClientMapperForCommand()).
+		NamespaceParam(cmdNamespace).DefaultNamespace().
+		FilenameParam(filenames...).
+		//SelectorParam(selector).
+		ResourceTypeOrNameArgs(true, args...).
+		Latest()
+	if err != nil {
+		return err
+	}
+
+	clientConfig, err := f.ClientConfig()
+	if err != nil {
+		return err
+	}
+
+	r := b.Flatten().Do()
```

Comment:
Since you don't make any other modifications to the builder, instead of doing this here, you should chain this on the resource builder, then immediatly check if there was an err from Do(). Then you need to check r.Err() to make sure there was no err.